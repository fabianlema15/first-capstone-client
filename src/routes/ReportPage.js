import React from 'react';
import GenericContext from '../contexts/GenericContext'
import ReportContent from '../components/Report/Report'
import ReportApiService from '../services/report-api-service';
import GenericApiService from '../services/generic-api-service';
import Helper from '../components/Utils/Helper';

class ReportPage extends React.Component{
  static defaultProps = {
    ROUTE : 'logins'
  }

  state = {
    userCode: null,
    userList: [],
    orderList: {},
    msgResponse: ''
  };

  componentDidMount() {
    GenericApiService.getAll('users')
      .then(result => {
        this.setState({userList: result.map(user => {
          return {
            id: user.id,
            value: user.last_name + ' ' + user.first_name
          }})})
      })
      .catch(this.setError)
  }

  setUserCode = userCode => {
    this.setState({ userCode })
  }


  setOrderList = orderList => {
    this.setState({ orderList })
  }

  setError = error => {
    this.setState({ error: error.error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  getReport = (e) => {
    e.preventDefault();
    const {user_id, from, to } = e.target;
    ReportApiService.getReport(user_id.value, from.value, to.value)
      .then(result => {
        console.log(result);
        this.setOrderList(Helper.serializeObj(result))
      })
      .catch(this.setError)
  }

  sendMail = (reportInfo) => {
    if (!Helper.objHasEmpty(reportInfo))
      ReportApiService.sendReportEmail(reportInfo)
      .then(result => {
        console.log(result);
        this.setState({
          msgResponse: 'Email sent'
        })
      })
      .catch(this.setError)
    else{
      this.setError({error: 'You need email'})
    }
  }

  render() {
    const value = {
      clearError: this.clearError,
      userCode: this.state.userCode,
      setUserCode: this.setUserCode,
      error: this.state.error,
      userList: this.state.userList,
      getReport: this.getReport,
      orderList: this.state.orderList,
      sendMail: this.sendMail,
      msgResponse: this.state.msgResponse,
      objName: 'Report'
    }

    return (
      <GenericContext.Provider value={value}>
        <ReportContent />
      </GenericContext.Provider>
    )
  }
}

export default ReportPage;
