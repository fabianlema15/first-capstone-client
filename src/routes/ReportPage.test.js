import React from 'react';
import ReactDOM from 'react-dom';
import ReportPage from './ReportPage'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';

describe('<ReportPage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
      <ReportPage>
      </ReportPage>
    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders ReportPage as expected', () => {
	    const tree = renderer.create(<BrowserRouter>
        <ReportPage>
        </ReportPage>
      </BrowserRouter>).toJSON();
	    expect(tree).toMatchSnapshot();
	  });

})
