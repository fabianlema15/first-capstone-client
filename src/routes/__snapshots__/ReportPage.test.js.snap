// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportPage /> Renders ReportPage as expected 1`] = `
<div>
  <nav>
    <div
      className="name-centered"
    >
      <h1>
        Report
      </h1>
    </div>
    <div
      className="back-nav"
    >
      <a
        href="/menu"
        onClick={[Function]}
      >
        Back
      </a>
    </div>
  </nav>
  <main
    className="content-with-nav"
  >
    <section>
      <form
        onSubmit={[Function]}
      >
        <h3>
          Reports
        </h3>
        <div
          className="fieldset"
        >
          <label
            htmlFor="from"
          >
            From
          </label>
          <input
            className="Input "
            id="from"
            onChange={[Function]}
            placeholder="Ex: 01/20/2019"
            required={true}
            type="date"
            value=""
          />
        </div>
        <div
          className="fieldset"
        >
          <label
            htmlFor="to"
          >
            To
          </label>
          <input
            className="Input "
            id="to"
            onChange={[Function]}
            placeholder="Ex: 12/31/2019"
            required={true}
            type="date"
            value=""
          />
        </div>
        <div
          className="fieldset"
        >
          <label
            htmlFor="user_id"
          >
            User
          </label>
          <select
            className="Textarea "
            id="user_id"
            onChange={[Function]}
            required={true}
            value=""
          >
            <option
              value=""
            >
              --SELECT ONE OPTION--
            </option>
          </select>
        </div>
        <button
          className="green"
          type="submit"
        >
          Get Report
        </button>
      </form>
    </section>
    <section>
      <div>
        You have no orders in this report
      </div>
    </section>
  </main>
</div>
`;
